### PR TITLE
[7.x] Add new `fromString` constructor to collections

### PR DESCRIPTION
**Pull request by JosephSilber**
This allows you to lazily iterate over the characters in a string:

```php
foreach (LazyCollection::fromString($hugeString) as $character) {
    //
}
```

### CODE REVIEW

File: src/Illuminate/Support/Enumerable.php
Comment (GrahamCampbell) [inline src/Illuminate/Support/Enumerable.php]: Breaking change.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/Enumerable.php
Comment (driesvints) [reply src/Illuminate/Support/Enumerable.php]: How so?

Comment (JosephSilber) [reply src/Illuminate/Support/Enumerable.php]: I guess if someone is implementing `Enumerable` themselves somewhere.

Comment (JosephSilber) [reply src/Illuminate/Support/Enumerable.php]: I removed it from the contract.

**Comment by GrahamCampbell** [comment]
I don't think we merge this without the contract changes (which you reverted). I was more meaning that we'd have to move this to `master`.

**Comment by JosephSilber** [comment]
> I don't think we merge this without the contract changes (which you reverted).

Totally disagree. It's very useful as is. No need to wait.

> I was more meaning that we'd have to move this to master.

After this is merged, I'll sent a PR to master adding it to the contract :+1:

**Comment by GrahamCampbell** [comment]
It is possible we can make a BC exemption since this is so early into the 7.0 release series. We could maybe treat it as a bug fix?

**Comment by GrahamCampbell** [comment]
Up to @taylorotwell.

**Comment by taylorotwell** [comment]
I'm gonna be honest and I know you'll probably ping me about this but I don't think this is a very common use case. Macro it in maybe?

**Pull request closed without merging**
